src/Simulation/Native/src/simulator/wavefunction.hpp

```diff
@@ -90,6 +91,52 @@ class Wavefunction
 
     void flush() const
     {
+        //logic to reorder
+        Fusion fg = fused_.get_fusedgates();
+        auto itemsToFuse = fg.get_items();
+        auto ctrlSet = fg.get_ctrl_set();
+        //getting all qubits to move to lower end of the wfn
+        if (itemsToFuse.size() > 0) {
```

Comment:
nit: `if (!itemsToFuse.empty())` (`empty()` is more likely to be O(1) than `size()`)